Sming/SmingCore/HardwarePWM.h

```diff
@@ -87,6 +102,10 @@ class HardwarePWM
      */
 	uint32 getMaxDuty();
 
+	/** @brief  This function is used to actually update the PWM.
+	 */
+	void updatePWM();
```

Comment:
Does this need to be a public method?

Reply:
Yes, it is needed in my opinion because after changing, let's say, duty of 5 channels (via `setDutyChan`) you need a way to activate the new duty values. Just setting `update` to `True` on the last channel you modified is not enough since it might get skipped (and therefore not updating PWM) if the new duty value is equal to the current one.

Reply:
OK, that's good. Maybe put that into the comment for `setDutyChan` ?

Reply:
@verybadsoldier Please, update the comment of  `setDutyChan` and specify that updatePWM() must be called in order to activate the new duty value. 

Another thing from my side - please, update the name of the `HardwarePWM::updatePWM` method to `HardwarePWM::update` and update the documentation and comments.

Once these two changes are done I think we are good to merge the PR.

Reply:
`updatePWM` does neccessarily has to be called after calling `setDutyChan`/`setDuty`. Please notice that `setDutyChan`/`setDuty` will auto-update the PWM by default. This is controlled by the (new) parameter `update` which defaults to `true`.

```
bool setDutyChan(uint8 chan, uint32 duty, bool update = true);
```
```
bool setDuty(uint8 pin, uint32 duty, bool update = true);
```

Calling `updatePWM` manually is only needed after calling one of those two set-Functions with `update=false` (and didn't call another set with `update=true` later).

Reply:
Sorry, added *NOT* in the first line in the last comment... pretty important...

Reply:
@verybadsoldier Ok, your explanation sounds good. Can you add it as a comment and change the name of the method to `update`? Once you push these changes and the CI system has checked it we will merge it so that we can release it this week with v3.7.1.